webrender/res/ps_image.vs.glsl

```diff
@@ -27,16 +27,28 @@ void main(void) {
 
     write_clip(vi.screen_pos, prim.clip_area);
 
+    vTileSpacing = image.stretch_size_and_tile_spacing.zw;
+    vStretchSize = image.stretch_size_and_tile_spacing.xy;
+
+    // We clamp the texture coordinates to the half-pixel offset from the borders
+    // in order to avoid sampling outside of the texture area.
     // vUv will contain how many times this image has wrapped around the image size.
+#ifndef WR_FEATURE_TEXTURE_RECT
     vec2 texture_size = vec2(textureSize(sColor0, 0));
```

Comment:
could be less of a code change if just `texture_size` is branched out, and the rest of the code is uniform